pkg/apis/apps/v1/defaults.go

```diff
@@ -26,6 +26,49 @@ func addDefaultingFuncs(scheme *runtime.Scheme) error {
 	return RegisterDefaults(scheme)
 }
 
+// SetDefaults_Deployment sets additional defaults compared to its counterpart
+// in extensions. These addons are:
+// - MaxUnavailable during rolling update set to 25% (1 in extensions)
+// - MaxSurge value during rolling update set to 25% (1 in extensions)
+// - RevisionHistoryLimit set to 10 (not set in extensions)
+// - ProgressDeadlineSeconds set to 600s (not set in extensions)
+func SetDefaults_Deployment(obj *appsv1.Deployment) {
+	// Set appsv1beta2.DeploymentSpec.Replicas to 1 if it is not set.
```

Comment:
nit: "appsv1"

Reply:
Good catch. I'm going to rebase the update into the first commit on the PR.